src/ValueResolver/NodeExpressionResolver.php

```diff
@@ -142,7 +142,7 @@ protected function resolveScalarString(Scalar\String_ $node)
     protected function resolveScalarMagicConstMethod()
     {
         if ($this->context instanceof \ReflectionMethod) {
-            $fullName = $this->context->getDeclaringClass()->getName() . '::' . $this->context->getShortName();
+            $fullName = $this->context->class . '::' . $this->context->getShortName();
```

Comment:
The suggestion in Scrutinizer CI inspection (see https://scrutinizer-ci.com/g/goaop/parser-reflection/inspections/eaf283f6-f982-44a4-bb28-51489bcba537/issues/files/src/ValueResolver/NodeExpressionResolver.php?status=fixed&orderField=path&order=asc&honorSelectedPaths=0) seems to be incorrect because mentioned https://bugs.php.net/bug.php?id=61384 PHP bug only affects `getName` method, but not `getDeclaringClass` method.

Knowing that I recommend

* replacing `$this->context->getDeclaringClass()->getName()`
* into `$this->context->getDeclaringClass()->name` instead.